packages/core/src/constants/chainId.ts

```diff
@@ -20,58 +54,9 @@ export enum ChainId {
   Avalanche = 43114,
 }
 
-export const CHAIN_NAMES = {
-  [ChainId.Mainnet]: 'Mainnet',
-  [ChainId.Ropsten]: 'Ropsten',
-  [ChainId.Kovan]: 'Kovan',
-  [ChainId.Rinkeby]: 'Rinkeby',
-  [ChainId.Goerli]: 'Goerli',
-  [ChainId.BSC]: 'BSC',
-  [ChainId.BSCTestnet]: 'BSCTestnet',
-  [ChainId.xDai]: 'xDai',
-  [ChainId.Polygon]: 'Polygon',
-  [ChainId.Theta]: 'Theta',
-  [ChainId.ThetaTestnet]: 'ThetaTestnet',
-  [ChainId.Moonriver]: 'Moonriver',
-  [ChainId.Mumbai]: 'Mumbai',
-  [ChainId.Harmony]: 'Harmony',
-  [ChainId.Palm]: 'Palm',
-  [ChainId.Localhost]: 'Localhost',
-  [ChainId.Hardhat]: 'Hardhat',
-  [ChainId.Fantom]: 'Fantom',
-  [ChainId.Avalanche]: 'Avalanche',
-}
-
-export const MULTICALL_ADDRESSES = {
-  [ChainId.Mainnet]: '0xeefba1e63905ef1d7acba5a8513c70307c1ce441',
-  [ChainId.Ropsten]: '0x53c43764255c17bd724f74c4ef150724ac50a3ed',
-  [ChainId.Rinkeby]: '0x42ad527de7d4e9d9d011ac45b31d8551f8fe9821',
-  [ChainId.Goerli]: '0x77dca2c955b15e9de4dbbcf1246b4b85b651e50e',
-  [ChainId.Kovan]: '0x2cc8688c5f75e365aaeeb4ea8d6a480405a48d2a',
-  [ChainId.BSC]: '0x41263cba59eb80dc200f3e2544eda4ed6a90e76c',
-  [ChainId.BSCTestnet]: '0xae11C5B5f29A6a25e955F0CB8ddCc416f522AF5C',
-  [ChainId.xDai]: '0xb5b692a88bdfc81ca69dcb1d924f59f0413a602a',
-  [ChainId.Polygon]: '0x11ce4B23bD875D7F5C6a31084f55fDe1e9A87507',
-  [ChainId.Theta]: '0xe2ec58a54f3ab2714eddbae87533793011f1e14e',
-  [ChainId.ThetaTestnet]: '0xf822bf2e728e264c58d7618022addd9cbc780350',
-  [ChainId.Moonriver]: '0xa9177F8d98DAaB74C24715Ba0A81b73654710523',
-  [ChainId.Mumbai]: '0x08411ADd0b5AA8ee47563b146743C13b3556c9Cc',
-  [ChainId.Harmony]: '0xFE4980f62D708c2A84D3929859Ea226340759320',
-  [ChainId.Palm]: '0x99a73dfE34578348fb81BD078201C0BA84E9c840',
-  [ChainId.Fantom]: '0xdc85396592f0F466224390771C861EE3957a3ff4',
-  [ChainId.Avalanche]: '0xdDCbf776dF3dE60163066A5ddDF2277cB445E0F3',
-}
-
-export const TEST_CHAINS = [
-  ChainId.Ropsten,
-  ChainId.Kovan,
-  ChainId.Rinkeby,
-  ChainId.BSCTestnet,
-  ChainId.Goerli,
-  ChainId.Mumbai,
-  ChainId.ThetaTestnet,
-  ChainId.Localhost,
-  ChainId.Hardhat,
-]
+export const MULTICALL_ADDRESSES: { [index: number]: string } = {}
+SUPPORT_NETWORKS.filter((network) => network.MULTICALL_ADDRESS).map(
+  (network) => (MULTICALL_ADDRESSES[network.chainId] = network.MULTICALL_ADDRESS)
+)
```

Comment:
I would propose to introduce a function `getMulticallAddressFor` to be used instead of MULTICALL_ADDRESSES constant, what do you think?

Reply:
And a helper function `getNetworkData(chainId)`

Reply:
Make sense, but it will break the backward compatibility for DAppProvider? Tend to scope this PR to refractory only (and no test change is required to scope the impact).

Does  the suggest `getNetworkData(chainId)` is used to return the full Chain config?

Reply:
Good point, it will break backward compatibility we should probably have a new name for it.
It is a good idea to keep scope of this PR to refactoring only, let's do that  👍 

Yes I would suggest to use getNetworkData(chainId) is used to return the full Chain config.